File: sdk/src/main/java/com/cxense/cxensesdk/model/ContentUser.java
package com.cxense.cxensesdk.model;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import java.util.HashMap;
import java.util.Map;

/**
 * This class describes user for server.
 *
 * @author Dmitriy Konopelkin (dmitry.konopelkin@cxense.com) on (2017-08-14).
 */

public class ContentUser {
    /**
     * Map from ID-types to IDs (String to String). ID-types are defined as a customer-prefix.
     */
    @SuppressWarnings({"UnusedDeclaration", "WeakerAccess"}) // Public API.
    @NonNull
    public final Map<String, String> ids;
    /**
     * User's likes
     */
    @SuppressWarnings({"UnusedDeclaration", "WeakerAccess"}) // Public API.
    @Nullable
    public UserPreference likes;
    /**
     * User's dislikes
     */
    @SuppressWarnings({"UnusedDeclaration", "WeakerAccess"}) // Public API.
    @Nullable
    public UserPreference dislikes;

    @SuppressWarnings({"UnusedDeclaration", "WeakerAccess"}) // Public API.
    public ContentUser() {
        ids = new HashMap<>();
    }

    @SuppressWarnings({"UnusedDeclaration", "WeakerAccess"}) // Public API.
    public ContentUser(@Nullable String userId) {
        this();
        ids.put("usi", userId);
    }
}

File: sdk/src/main/java/com/cxense/cxensesdk/model/ContentUser.kt
package com.cxense.cxensesdk.model

/**
 * This class describes user for server.
 *
 */
@Suppress("unused") // Public API.
class ContentUser() {
    constructor(userId: String) : this() {
        ids[USI_ID] = userId
    }

    /**
     * Map from ID-types to IDs (String to String). ID-types are defined as a customer-prefix.
     */
    val ids: MutableMap<String, String> = mutableMapOf()
    /**
     * User's likes
     */
    var likes: UserPreference? = null
    /**
     * User's dislikes
     */
    var dislikes: UserPreference? = null

    companion object {
        const val USI_ID = "usi"
    }
}
